File: CPP/6ZigZagConversion.cpp
#include <iostream>
#include <cstring>

using namespace std;
class Solution {
public:
    string convert(string s, int numRows) {
    	if(numRows <=1 || s.size() ==0)
			return s;
		string ans="";
		int n= s.size();
		for(int i=0; i<n&&i<numRows; i++){
			int index = i;
			ans += s[index];
			for(int j=1; index<n; j++){
				if(i==0 || i==numRows-1){
					index += 2*numRows -2;
				}else{
					if(j%2!=0){
						index += 2*(numRows-1-i);
					}else{
						index += 2*i;
					}
				}
				
				if(index < n){
					ans += s[index];
				}
			}
		}
		return ans;    
    }
};


int main(){
	return 0;
} 

File: Python/6ZigZagConversion.py
__author__ = 'Minyi'
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if len(s)==0 or numRows==1:
            return s
        ans = ''
        i = 0;
        n = len(s)
        while (i<n and i<numRows):
            index = i
            ans += s[index]
            j = 1
            while index < n:
                if (i==0 or i==numRows-1):
                    index += 2*numRows-2
                else:
                    if j%2!=0:
                        index += 2*(numRows-1-i)
                    else:
                        index += 2*i
                if index < n:
                    ans += s[index]

                j += 1
            i += 1
        return ans